README.md

```diff
@@ -430,6 +430,132 @@ public function getUser($credentials, UserProviderInterface $userProvider)
 The logged-in user will be an instance of `KnpU\OAuth2ClientBundle\Security\User\OAuthUser` and will
 have the roles `ROLE_USER` and `ROLE_OAUTH_USER`.
 
+## Using the new Symfony Authenticator
+Now you can use the new Symfony Authenticator system (available **since Symfony 5.2**, don't you use it before this version) to login in your app.
+### Step 1) Using the new Oauth2Authenticator class
+```php
+<?php
+
+
+namespace App\Security;
+
+use App\Entity\User; // your user entity
+use Doctrine\ORM\EntityManagerInterface;
+use KnpU\OAuth2ClientBundle\Client\ClientRegistry;
+use KnpU\OAuth2ClientBundle\Security\Authenticator\OAuth2Authenticator;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Http\Authenticator\Passport\PassportInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\SelfValidatingPassport;
+
+class MyFacebookAuthenticator extends OAuth2Authenticator
+{
+    private $clientRegistry;
+    private $entityManager;
+    private $router;
+
+    public function __construct(ClientRegistry $clientRegistry, EntityManagerInterface $entityManager, RouterInterface $router)
+    {
+        $this->clientRegistry = $clientRegistry;
+        $this->entityManager = $entityManager;
+        $this->router = $router;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function supports(Request $request): ?bool
+    {
+        // continue ONLY if the current ROUTE matches the check ROUTE
+        return $request->attributes->get('_route') === 'connect_facebook_check';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function authenticate(Request $request): PassportInterface
+    {
+        $credentials = $this->fetchAccessToken($this->clientRegistry->getClient('facebook_main'));
+
+        return new SelfValidatingPassport(
+            new UserBadge($credentials, function($credentials) {
+                /** @var FacebookUser $facebookUser */
+                $facebookUser = $this->clientRegistry->getClient('facebook_main')->fetchUserFromToken($credentials);
+
+                $email = $facebookUser->getEmail();
+
+                // 1) have they logged in with Facebook before? Easy!
+                $existingUser = $this->em->getRepository(User::class)->findOneBy['facebookId' => $facebookUser->getId()];
```

Comment:
You are missing the brackets

`$existingUser = $this->em->getRepository(User::class)->findOneBy(['facebookId' => $facebookUser->getId()]);`